frontend/docsettings.lua

```diff
@@ -424,18 +490,24 @@ local function findCoverFileInDir(dir)
     end
 end
 
---- Returns path to book custom cover file if it exists, or nil.
+--- Returns path to book custom cover file if it exists, or nil
 function DocSettings:findCoverFile(doc_path)
     doc_path = doc_path or self.data.doc_path
     local location = G_reader_settings:readSetting("document_metadata_folder", "doc")
     local sidecar_dir = self:getSidecarDir(doc_path, location)
     local cover_file = findCoverFileInDir(sidecar_dir)
-    if not cover_file then
-        location = location == "doc" and "dir" or "doc"
-        sidecar_dir = self:getSidecarDir(doc_path, location)
-        cover_file = findCoverFileInDir(sidecar_dir)
+    if cover_file then return cover_file end
+    local candidates = {"doc", "dir"}
+    if DocSettings.isHashLocationEnabled() then
+        table.insert(candidates, "hash")
+    end
```

Comment:
No specific ordering needed here?
Also, there seems to be a first check (remnant of when we only had 2 checks), and then a loop with additional checks including the first check ? Can't it be just the loop?

Reply:
I think @hius07 is saying the code is okay as is here?

Reply:
No it isn't, the preferred location is checked twice.

Reply:
Sorry, I'm blindly not seeing what you mean by checked twice if `candidates` is iterated over once.

Reply:
Yes, it's okay.